core/vdbe/mod.rs

```diff
@@ -360,6 +360,25 @@ impl Program {
                     state.registers[*dest] = exec_bit_not(&state.registers[*reg]);
                     state.pc += 1;
                 }
+                Insn::Checkpoint {
+                    database: _,
+                    checkpoint_mode: _,
+                    dest,
+                } => {
+                    let result = self.connection.upgrade().unwrap().checkpoint();
+                    match result {
+                        Ok(()) => {
+                            // fixme currently checkpoint result are hard-coded
+                            // 1st register: 1 (checkpoint SQLITE_BUSY) or 0 (not busy).
+                            state.registers[*dest] = OwnedValue::Integer(0);
+                            state.registers[*dest + 1] = OwnedValue::Integer(0);
+                            state.registers[*dest + 2] = OwnedValue::Integer(0);
```

Comment:
todo

- more comments here: 2nd, 3rd registers; number of pages in wal and checkpointed are not yet supported
- 1st output register based on checkpoint result instead of hard-coded